guides/plugins/plugins/framework/data-handling/field-inheritance.md

```diff
@@ -0,0 +1,227 @@
+# Field inheritance
+
+## Overview
+
+In this guide you'll learn how to create inherited fields for your entities.
+
+## Prerequisites
+
+This guide is built upon the [Plugin Base Guide](../../plugin-base-guide.md), but any plugin will work here. Just note that all examples are using the plugin mentioned above.
+
+You also should have a look at our [Adding custom complex data](./add-custom-complex-data.md) guide, since this guide is built upon it.
+
+## Inherit a field
+
+To start using inheritance, we have to update our definition and database.
+
+1. Make inheritable fields nullable in the database
+2. Add the `ParentFkField`, `ParentAssociationField`, `ChildrenAssociationField` in your definition
+3. Enable inheritance by overwriting `isInheritanceAware()`
+4. Flag fields as inheritable
+5. Add getters and setters to the entity class
+
+### 1. Make fields nullable
+
+The first thing we need to do is to make all our fields that we want to make inheritable nullable in our migration.
+
+```sql
+ALTER TABLE `swag_example` MODIFY `description` VARCHAR(255) NULL;
+```
+
+### 2. Add the ParentFkField and the associations
+
+After we've made all our fields nullable, we still need to add the following fields to our definition:
+`Shopware\Core\Framework\DataAbstractionLayer\Field\ParentFkField`, `Shopware\Core\Framework\DataAbstractionLayer\Field\ParentAssociationField` 
+and `Shopware\Core\Framework\DataAbstractionLayer\Field\ChildrenAssociationField`.
```

Comment:
Please just do not explain what to do here. Try to explain, what these Fields do:

ParentFkField is the foreign key, that references the parent's id. In default, it points to a `parent_id` column in the database.

Please do this for the ParentAssociationField and ChildrenAssociationField aswell

Reply:
Agree. Should definitely explain what the fields do